src/TemplateString.php

```
@@ -0,0 +1,230 @@
+<?php
+
+namespace ipl\Html;
+
+/**
+ * Render {{#mustache}}Mustache{{/mustache}}-like string from {@link ValidHtml} element arguments
+ *
+ * # Example Usage
+ * ```
+ * $info = new TemplateString(
+ *     'Follow the %s for more information on %s {{#mustache}}Mustache{{/mustache}}',
+ *     [
+ *         new Link('doc/html', 'HTML documentation'),
+ *         Html::tag('strong', 'HTML elements'),
+ *        [
+ *        'mustache' => Html::tag('span')
+ *        ]
+ *     ]
+ * );
+ * ```
+ */
+class TemplateString extends FormattedString
+{
+    /** @var array */
+    protected $templateArgs;
+
+    protected $pos;
+
+    /** @var int Length of {{#}} | {{/}}*/
+    const MUSTACHE_LENGTH = 5;
+
+    public function __construct($format, $args = null)
+    {
+        $this->templateArgs = [];
+
+        foreach ($args as $key => $val) {
+            if (is_array($val) && $this->isAssoc($val)) {
```

I believe you can replace `$this->isAssoc($val)` with `is_string(key($val))`. If the first key isn't a string the array isn't a template argument. If there are indeed other keys which are not strings, the rendering should fail in my opinion. String keys don't matter for `FormattedString` anyway, so a developer shouldn't mix key types in any case. And if it's done, it should throw an error instead of silently accepting it.